### PR TITLE
[3006.x] update photon paths to use their $releasever string

### PR DESCRIPTION
**Pull request by felippeb**
### What does this PR do?
fix photon3/4/5 repo paths

### CODE REVIEW

File: tests/pytests/pkg/download/test_pkg_download.py [hunk 1]
```diff
@@ -256,6 +256,9 @@ def setup_redhat_family(
 ):
     arch = os.environ.get("SALT_REPO_ARCH") or "x86_64"
 
+    if os_name == "photon":
+        os_version = f"{os_version}.0"
```
Comment (dmurphy18) [inline tests/pytests/pkg/download/test_pkg_download.py:260]: Wondering what happens when '.1' comes along

File: tools/pkg/repo/create.py [hunk 2]
```diff
@@ -380,6 +380,9 @@ def rpm(
         assert incoming is not None
         assert repo_path is not None
         assert key_id is not None
+
+    if distro == "photon":
+        distro_version = f"{distro_version}.0"
```
Comment (dmurphy18) [inline tools/pkg/repo/create.py:385]: can we always rely on '.0' , probably not

**Review by dmurphy18 — CHANGES_REQUESTED** [review]

File: tests/pytests/pkg/download/test_pkg_download.py (hunk 1 above)
Comment (s0undt3ch) [reply tests/pytests/pkg/download/test_pkg_download.py:260]: It doesn't, never has.

Comment (s0undt3ch) [reply tests/pytests/pkg/download/test_pkg_download.py:260]: But @felippeb knows more about that information.

Comment (felippeb) [reply tests/pytests/pkg/download/test_pkg_download.py:260]: The photon team has indicated that they have never used that versioning. they put a .X in place if they ever wanted to, but functionally never have.

File: tools/pkg/repo/create.py (hunk 2 above)
Comment (felippeb) [reply tools/pkg/repo/create.py:385]: The photon team has indicated that they have never used that versioning. they put a .X in place if they ever wanted to, but functionally never have.

**Review by dmurphy18 — APPROVED** [review]

**Pull request merged**